include/fastrtps/participant/Participant.h

```diff
@@ -86,6 +87,11 @@ class RTPS_DllAPI Participant
     bool get_remote_writer_info(const rtps::GUID_t& writerGuid, rtps::WriterProxyData& returnedInfo);
 
     bool get_remote_reader_info(const rtps::GUID_t& readerGuid, rtps::ReaderProxyData& returnedInfo);
+
+    /**
+     * Get the resource event for this participant
+     */
+    rtps::ResourceEvent& get_resource_event() const;
```

Comment:
Why is it needed at public API level?

Reply:
As in the deadline case, I must have thought it was needed at some point but I've removed it now.